extends HBoxContainer


# Called when the node enters the scene tree for the first time.
func _ready() -> void:
	pass # Replace with function body.


# Called every frame. 'delta' is the elapsed time since the previous frame.
func _process(_delta: float) -> void:
	pass


func _on_button_pressed() -> void:
	var button_text:Array[String]=["æ£€æŸ¥å¯åŠ¨å™¨æ›´æ–°","æ£€æŸ¥æ¸¸æˆæœ¬ä½“æ›´æ–°","è®¿é—®ä»“åº“","ç¡®å®š"]
	var detail="ç”±äºæŠ€æœ¯æœ‰é™ï¼Œå‡ºç°æœªçŸ¥å¼‚å¸¸å¯å…ˆè”ç³»å¼€å‘è€…ç¡®è®¤æ˜¯å¦ä¸ºBUG\nå¦‚æœéœ€è¦è‡ªè¡Œè§£å†³ï¼Œå¯ä¸‹è½½æœ€æ–°ç‰ˆï¼Œé‡è£…è§£å†³ğŸŒ¹"
	var result=await WindowsManager.show_dialog("æç¤º",detail, button_text)
	
	if result=="æ£€æŸ¥å¯åŠ¨å™¨æ›´æ–°":
		WindowsManager.version_check(1)
		return
	if result=="æ£€æŸ¥æ¸¸æˆæœ¬ä½“æ›´æ–°":
		WindowsManager.version_check(2)
		return
	
	if result=="è®¿é—®ä»“åº“":
		var button_text_1:Array[String]=["å¯åŠ¨å™¨","æ¸¸æˆæœ¬ä½“","å–æ¶ˆ"]
		var result_0=await WindowsManager.show_dialog("é€‰æ‹©ä»“åº“","ä½ å¯ä»¥é€‰æ‹©ä¸‹é¢çš„ä»“åº“ç±»å‹ï¼Œä½ å°†ä¼šç›´æ¥åˆ°è¾¾æ¸¸æˆå‘è¡Œç‰ˆçš„ä¸‹è½½é¡µé¢ï¼Œä½ å¯ä»¥é€‰æ‹©ç›´æ¥ä¸‹è½½æœ€æ–°ç‰ˆ",button_text_1)
		if result_0=="å¯åŠ¨å™¨":
			OS.shell_open("https://github.com/GlacierXiaowei/ETN_Launcher/releases")
			return
		if result_0=="æ¸¸æˆæœ¬ä½“":
			OS.shell_open("https://github.com/GlacierXiaowei/ETN/releases")
			return
